src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -3150,6 +3173,8 @@ void processContextElement
     return; // Error already in responseP
   } 
 
+  cacheSemTake(__FUNCTION__, "Processing Context Element");
```

Comment:
Thinking in the new location of the semaphores, let's take into account that processContextElement is a bit bunch of code, almost all the logic from updating each CE is there.

So, while update context is being processes other update context will be blocked. It is like going back to the old `-reqMutextPolicy all` days. Even worse: I understand that the cache update thread and subscription/update subscription operation will also block...

Not sure what to do regarding this. Maybe we should talk at skype.

(Sorry, probably I should have detected this when you told me about the changes in the sempahore strategy this morning).